src/arrayMethodReduce.js

```diff
@@ -5,7 +5,18 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    const hasInitialValue = arguments.length > 1;
+    let prev = initialValue;
+
+    for (let i = 0; i < this.length; i++) {
+      if (i === 0 && !hasInitialValue) {
+        prev = this[i];
+      } else {
+        prev = callback(prev, this[i], i, this);
+      }
```

Comment:
you can avoid using this validation by checking for params before the loop
```
if (args.length < 2) {
...
}

for ...

```
you can look up approaches of your peers